### PR TITLE
Update vault docs

### PR DESCRIPTION
**Pull request by josefkarasek**
## Description

Changes proposed in this pull request:

- Update vault example to work with socket slack communicator

## Testing

Follow the steps in the doc

## Related issue(s)

https://github.com/kubeshop/botkube-docs/issues/193

### CODE REVIEW

File: docs/configuration/communication/vault-csi.md [hunk 1]
```diff
@@ -96,33 +100,43 @@ This instruction guides you through the installation of Botkube and Vault on a K
...
      # Settings for Slack
-     slack:
+     socketSlack:
        enabled: true
        channel: 'random'
        # token - specified via env variable
 
    extraEnv:
-     - name: COMMUNICATIONS_SLACK_TOKEN
+     - name: COMMUNICATIONS_SOCKETSLACK_APPTOKEN
```
Comment (josefkarasek) [inline docs/configuration/communication/vault-csi.md:130]: `COMMUNICATIONS_SOCKETSLACK_APPTOKEN` is not correct. Maybe some extra `_`?

File: docs/configuration/communication/vault-csi.md [hunk 2]
```diff
@@ -44,7 +44,8 @@ This instruction guides you through the installation of Botkube and Vault on a K
```
Comment (pkosiec) [inline docs/configuration/communication/vault-csi.md:44]: Can you also propagate the changes to Botkube 0.16 and up? Thanks in advance!

File: docs/configuration/communication/vault-csi.md [hunk 3]
```diff
@@ -44,7 +44,8 @@ This instruction guides you through the installation of Botkube and Vault on a K
 
    ```bash
    # Write the token to Vault
-   vault kv put secret/slack token={token}
+   vault kv put -mount=secret slack-app-token token={xapp-...}
+   vault kv put -mount=secret slack-bot-token token={xoxb-...}
```
Comment (pkosiec) [inline docs/configuration/communication/vault-csi.md:48]: Just to make it clear, I would suggest going in one or another direction:

Option 1:
```suggestion
   vault kv put -mount=secret slack-app-token token=xapp-...
   vault kv put -mount=secret slack-bot-token token=xoxb-...
```

Option 2:
```suggestion
   vault kv put -mount=secret slack-app-token token={appToken}
   vault kv put -mount=secret slack-bot-token token={botToken}
```

Whatever you prefer 🙂

**Review by pkosiec — APPROVED** [review]
LGTM with a very small comments. Good work! 
Disclaimer: Didn't test it, but I know you did, so 👌

File: docs/configuration/communication/vault-csi.md [hunk 4]
```diff
@@ -96,33 +100,43 @@ This instruction guides you through the installation of Botkube and Vault on a K
...
+             key: "slack-app-token"
+           - objectName: "slack-bot-token"
+             key: "slack-bot-token"
 
    communications:
      # Settings for Slack
-     slack:
+     socketSlack:
        enabled: true
        channel: 'random'
```
Comment (pkosiec) [inline docs/configuration/communication/vault-csi.md:126]: This is outdated - please fix it 👍

**Review by pkosiec — APPROVED** [review]
LGTM

**Pull request merged**